simapp/test_helpers.go

```diff
@@ -92,9 +93,13 @@ func SetupWithGenesisValSet(t *testing.T, valSet *tmtypes.ValidatorSet, genAccs
 	bondAmt := sdk.NewInt(1000000)
 
 	for _, val := range valSet.Validators {
+		pk, err := ed25519.FromTmEd25519(val.PubKey)
```

Comment:
Could we add a comment here that we're right now only supporting ed25519 as TM pubkey.

Tracking https://github.com/cosmos/cosmos-sdk/pull/7604 too, as we might need a switch type here.

Reply:
I've updated the issue and added the link to this code.